src/PkgAuthentication.jl

```diff
@@ -263,47 +292,52 @@ function get_server_dir(url::AbstractString, server=Pkg.pkg_server())
     joinpath(Pkg.depots1(), "servers", domain)
 end
 
-function token_path(url)
+function token_path(url::AbstractString)
     if is_new_auth_mechanism()
         server_dir = get_server_dir(url)
         if server_dir !== nothing
             return joinpath(server_dir, "auth.toml")
         end
     end
-    get(ENV, "JULIA_PKG_TOKEN_PATH", joinpath(homedir(), ".julia", "token.toml"))
+    # older auth mechanism uses a different token location
+    # QUESTION: do we still need this for Julia ≥ 1.3?
+    default = joinpath(Pkg.depots1(), "token.toml")
+    get(ENV, "JULIA_PKG_TOKEN_PATH", default)
 end
 
-const OPEN_BROWSER_HOOK = Ref{Any}(nothing)
+const OPEN_BROWSER_HOOK = Ref{Union{Function, Nothing}}(nothing)
 
-function register_open_browser_hook(f)
+function register_open_browser_hook(f::Function)
+    if !hasmethod(f, Tuple{AbstractString})
+        throw(ArgumentError("Browser hook must be a function taking a single URL string argument."))
+    end
     OPEN_BROWSER_HOOK[] = f
 end
 
 function clear_open_browser_hook()
     OPEN_BROWSER_HOOK[] = nothing
 end
 
-function open_browser(url)
+function open_browser(url::AbstractString)
+    @info "opening auth in browser"
```

Comment:
Should probably get rid of this.

Reply:
Is the other info logging I'm seeing coming from the tests? We should either consistently log info or not. I tweaked the output here to try to make it as informative to the user as possible so that in the case that their browser is not configured to bring a new window to the front or opening the browser fails for some reason they at least have some recourse.